plutus-core/plutus-core/satint-test/TestSatInt.hs

```diff
@@ -53,7 +53,9 @@ tests =
     unitTest "min-"    (saturatesNeg ((minBound :: SatInt) - 1)),
     unitTest "1/0"     (isArithException (1 `div` 0)),
     unitTest "min*-1"  (saturatesPos ((minBound :: SatInt) * (-1))),
-    unitTest "min/-1"  (saturatesNeg ((minBound :: SatInt) `div` (-1))),
+    unitTest "0-min"   (saturatesPos (0 - (minBound :: SatInt))),
```

Comment:
I experimented with the property tests a bit. First of all, they weren't using a very good generator, so I fixed that, which led to lots of div by zero errors (because it wasn't picking 0 before!), so I fixed the handling for that. Then I broke the implementation again, but even with ratcheting up the number of tests, it didn't pick `minBound` and `-1` ever. Eventually I wrote a manual generator that prefers the edge cases and then I managed to make it break. In the process I found out that `gcd` and `lcm` are fundamentally broken though, which is good!

Reply:
>  Eventually I wrote a manual generator that prefers the edge cases

I had to do something very similar with the tests I wrote for the `upperIntegerQuotient` function a couple of days ago.  I used Hedgehog since that's what was already being used for the rest of the tests in `plutus-core` but the default generators didn't seem to do a very good job.  There's a `Size` parameter involved that changes hoe much of a range is covered, but the documentation doesn't really describe how that's used (I think maybe it's involved shrinking).  Experimentation showed that if you were asking for samples from [a,b] it would only choose numbers from the lower 30% with the default `Size` value.  Eventually I had to write special generators for both ends and the middle of the `SatInt` range.  I suppose the moral is "check your generators".